project-fortis-interfaces/src/components/Insights/Maps/HeatMap.js

```diff
@@ -75,10 +75,14 @@ export default class HeatMap extends React.Component {
 
   componentWillReceiveProps(nextProps) {
     const { placeid, defaultZoom } = this.state;
+    const { targetBbox } = this.props;
 
     if (hasChanged(this.props, nextProps) && nextProps.selectedplace.placeid && placeid !== nextProps.selectedplace.placeid) {
       this.moveMapToNewLocation(nextProps, defaultZoom);
     }
+    if (hasChanged(this.props, nextProps) && nextProps.bbox && JSON.stringify(nextProps.bbox) === JSON.stringify(targetBbox)) {
```

Comment:
Do we need both of these if statements or can they be combined? I.e. is there ever a case where it would make sense to first move to a particular location and then to a bounding box? If not, consider merging the if statements to make it explicit that they are mutually exclusive operations.

```js
if (hasChanged(this.props, nextProps)) {
  if (nextProps.selectedplace.placeid && placeid !== nextProps.selectedplace.placeid) {
    this.moveMapToNewLocation(nextProps, defaultZoom);
  } else if (nextProps.bbox && JSON.stringify(nextProps.bbox) === JSON.stringify(targetBbox)) {
    this.moveMapToBoundingBox(targetBbox);
  }
}
```